.travis.yml

```diff
@@ -10,12 +15,27 @@ env:
     - TASK=build
     - TASK=mpi-build
     - TASK=cmake-build
-    - TASK=test CXX=g++-4.8
+    - TASK=test CXX=g++
+    - TASK=xgb-java-tests
+
+matrix:
+  exclude:
```

Comment:
why these are excluded?

Reply:
https://github.com/chenqin/rabit/blob/master/.travis.yml#L22
exclude osx lint and doc as well as mpi, for first three since we already do in linux so it's matter of save build time. For mpi, I don't see any use case running mpi in osx, we do have linux tests running btw.

Reply:
this is concerning, we have seen enough cases about different network behaviors in Mac/Linux

As a developer, test behaviors in mac os before deploying to linux production servers is a common behavior

I would suggest we pursue the consistency of tests in different OS with the best effort